File: uva/488.cpp
#include <stdio.h>
#include <string>
#include <iostream>

using namespace std;

string seq[10], wave[10], num="123456789";

int main(){
    for (int i=1; i<10; i++){
        seq[i]="";
        for (int j=0; j<i; j++) seq[i]=seq[i]+num[i-1];
        wave[i]="";
        for (int j=1; j<=i; j++) wave[i]=wave[i]+seq[j]+"\n";
        for (int j=i-1; j>0; j--) wave[i]=wave[i]+seq[j]+"\n";
        }
    int tc, f, a;
    cin >> tc;
    for (int kasus=0; kasus<tc; kasus++){
        if (kasus>0) cout << endl;
        cin >> a >> f;
        for (int i=1; i<f; i++) cout << wave[a] << endl;
        cout << wave[a];
        }
}

File: uva/488.pas
const
  wave:array[1..9] of string = ('1','22','333','4444','55555','666666','7777777','88888888','999999999');

var
  tc,kasus,A,f,i,j:longint;

begin
  readln(tc);
  for kasus:=1 to tc do
    begin
	  readln(A); readln(f);
	  for i:=1 to f do
	    begin
		  for j:=1 to A do writeln(wave[j]);;
		  for j:=A-1 downto 1 do writeln(wave[j]);
		  if i<f then writeln;
		end;
	  if kasus<tc then writeln;
	end;
end.
